src/MapInterface.php

```diff
@@ -0,0 +1,18 @@
+<?php
+
+namespace Dhii\Collection;
+
+use Psr\Container\ContainerInterface;
+
+/**
+ * A readable set.
+ *
+ * @since [*next-version*]
+ */
+interface MapInterface extends
```

Comment:
Read my comments about `SetInterface` first.

Given my opinion on sets, my opinion here is that a map should be a `ContainerInterface` and a `Traversable`.